File: task-manager-backend/src/main/java/com/example/task_manager_backend/service/TaskService.java
package com.example.task_manager_backend.service;

import com.example.task_manager_backend.DTO.TaskDto;

import java.util.List;

public interface TaskService {

    TaskDto createTask(TaskDto taskDto);
    TaskDto getTaskById(Long id);
    List<TaskDto> getAllTasks();
    TaskDto updateTask(Long id, TaskDto updatedTask);
    void deleteTask(Long id);

}

File: task-manager-frontend/src/app/services/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';


export interface Task {
  id?: number;
  title: string;
  description: string;
  status: string;
}

@Injectable({
  providedIn: 'root'
})
export class TaskService {
  private baseUrl = 'http://localhost:8091/api/v1/tasks';

  constructor(private http: HttpClient) { }

  getAllTasks(): Observable<Task[]> {
    return this.http.get<Task[]>(this.baseUrl);
  }

  createTask(task: Task): Observable<Task> {
    return this.http.post<Task>(this.baseUrl, task);
  }

  deleteTask(id: number): Observable<string> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }
  getTaskById(id: number): Observable<Task> {
  return this.http.get<Task>(`${this.baseUrl}/${id}`);
}

updateTask(id: number, task: Task): Observable<Task> {
  return this.http.put<Task>(`${this.baseUrl}/${id}`, task);
}
}